### PR TITLE
Components: Adds LoggedOutForm component

### PR DESCRIPTION
**Pull request by ebinnion**
In an effort to factor the logged out link that shows at the bottom of the form at `/start/account/user` and to allow reuse of the logged out form styles, this PR refactors the `signup/logged-out-form` component and brings much of its functionality to `client/components/logged-out-form-container` component.

cc @mtias and @rickybanister for design review, @lezama, and @drewblaisdell for code review.

To test:
- Checkout `add/components-logged-out-form` branch
- Test invite flows
  - On a WP.com site, invite a test user by going to `$site/wp-admin/users.php?page=wpcom-invite-users`
  - In the invite email, make note of the invitation key
  - Go to `/accept-invite/$site/$invite_key`
  - While logged in or out, can you accept an invite? Are there errors? Does everything look OK?
- Test user creation
  - Go to `/start/account/user`
  - Attempt to create a user
  - Does everything work and look good?
- Test site creation
  - Go to `/start/site-user/site`
  - Does everything look ok? Are there any errors?

__Note:__ You'll notice that the `/log-in` form is off centered, and during testing you will likely run into a JS error if you try to submit the form on `/log-in`. This is not related to this PR and `/log-in` is only enabled in `development`.

__Also note:__ In some screenshots you'll notice there's quite a bit of padding between the form footer and the input directly above it. This seems to be caused by:

```
.validation-fieldset + .logged-out-form__footer {
    margin-top: 20px;
}
```

I intentionally added a screenshot with an invalid form field so that you could see what that would look like.

__After screenshots:__
screen shot
screen shot 1
screen shot 2
screen shot 3
screen shot 4
screen shot

### CODE REVIEW

File: client/components/logged-out-form/footer.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import React from 'react';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+
+module.exports = React.createClass( {
+	displayName: 'LoggedOutFormFooter',
```
Comment (enejb) [inline client/components/logged-out-form/footer.jsx]: missing propTypes: {} The same is true for the rest of the new components.

File: client/components/logged-out-form/link-item.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * External dependencies
+ */
+import React from 'react';
+import classnames from 'classnames';
+import omit from 'lodash/object/omit';
+
+module.exports = React.createClass( {
+	displayName: 'LoggedOutFormLinkItem',
+
+	render: function() {
```
Comment (enejb) [inline client/components/logged-out-form/link-item.jsx]: could be done as es6 
so
`render() {` 

and `module.exports =` should be `export default`

**Comment by enejb** [comment]
I left some minor comments. Overall in my testing I didn't find any errors beside the once that you pointed out already. I was able to accept the invitation + create a new user and a new site.

**Comment by ebinnion** [comment]
@enejb I went ahead and `ES6-ified` and added `propTypes` to address your comments.

**Comment by ebinnion** [comment]
Added missing `border-top` to the form footer.

screen shot

**Comment by rickybanister** [comment]
What's the email-address-only 'log me in' form for?

The footer looks good in all the examples. This seems like a useful refactor.

**Comment by ebinnion** [comment]
@rickybanister I'm not entirely sure. I think it was an attempt at mobile login, but it hasn't launched yet.

**Comment by lezama** [comment]
This looks much cleaner and ready to go to me :)

**Pull request merged**